server/src/main/java/org/opensearch/extensions/ExtensionsSettings.java

```diff
@@ -184,6 +186,10 @@ public void setHasNativeController(String hasNativeController) {
             this.hasNativeController = hasNativeController;
         }
 
+        public List<ExtensionDependency> getDependency() {
```

Comment:
(Style) Unless there's a really good reason not to do so, the name of a getter should match the field.  Since we are dealing with a list of dependencies, this should be `getDependencies()`.